jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -252,6 +252,7 @@ public void setTempDirectory(File tempDirectory)
             }
         }
         _tempDirectory = tempDirectory;
+         _createdTempDirectoryName = false;
```

Comment:
You need to delete any directory that had been created (but only other name is different to what is set)

Reply:
Done.